lib/internal/streams/readable.js

```diff
@@ -162,7 +163,14 @@ function ReadableState(options, stream, isDuplex) {
   // Crypto is kind of old and crusty.  Historically, its default string
   // encoding is 'binary' so we have to make this configurable.
   // Everything else in the universe uses 'utf8', though.
-  this.defaultEncoding = (options && options.defaultEncoding) || 'utf8';
+  const defaultEncoding = options?.defaultEncoding;
+  if (defaultEncoding == null) {
+    this.defaultEncoding = 'utf8';
+  } else if (Buffer.isEncoding(defaultEncoding)) {
+    this.defaultEncoding = defaultEncoding;
+  } else {
+    throw new ERR_UNKNOWN_ENCODING(defaultEncoding);
+  }
```

Comment:
Perhaps... 
```suggestion
  const defaultEncoding = options?.defaultEncoding || 'utf8';
  if (!Buffer.isEncoding(defaultEncoding))
    throw new ERR_UNKNOWN_ENCODING(defaultEncoding);
  this.defaultEncoding = defaultEncoding;
```

Reply:
With your proposed change, if null or undefined we would assign it to default value and then check `Buffer.isEncoding` which is redundant in my opinion.